finesse/gui/temp_control.py

```diff
@@ -0,0 +1,359 @@
+"""Panel and widgets related to temperature monitoring."""
+
+import matplotlib.pyplot as plt
+from matplotlib.backends.backend_qt5agg import FigureCanvasQTAgg
+from PySide6.QtCore import Qt
+from PySide6.QtGui import QPixmap
+from PySide6.QtWidgets import (
+    QGridLayout,
+    QGroupBox,
+    QLabel,
+    QLineEdit,
+    QPushButton,
+    QSlider,
+    QSpinBox,
+    QWidget,
+)
+
+# plt.style.use("./finesse_gui.style")
+plt.rcParams["figure.facecolor"] = "black"
+plt.rcParams["axes.facecolor"] = "black"
+plt.rcParams["axes.edgecolor"] = "white"
+plt.rcParams["text.color"] = "white"
+plt.rcParams["xtick.color"] = "white"
+plt.rcParams["ytick.color"] = "white"
+
+
+class BBMonitor(QGroupBox):
+    """Widgets to view the temperature properties."""
+
+    def __init__(self) -> None:
+        """Creates a panel with a graph to monitor the hot and cold bb temperatures."""
+        super().__init__("BB Monitor")
+
+        layout = self._create_controls()
+        self.setLayout(layout)
+
+    def _create_controls(self) -> QGridLayout:
+        """Creates the overall layout for the panel.
+
+        Returns:
+            QGridLayout: The layout containing the figure.
+        """
+        layout = QGridLayout()
+        self._create_figure()
+
+        layout.addWidget(self._canvas, 0, 0)
+
+        return layout
+
+    def _create_figure(self) -> None:
+        """Creates the matplotlib figure to be contained within the panel.
+
+        Returns:
+            None
+        """
+        self._figure, self._ax1 = plt.subplots()
```

Comment:
I would keep the axes in a dictionary, so they can be updated based on a signal coming from somewhere in an easier way:
```suggestion
        self._figure, ax = plt.subplots()
        self.ax = {"hot": ax}
        ...
        self.ax["cold"] = self.ax["hot"].twinx()
```

And a similar thing with the lines, putting them into `self.lines["hot"]` and `self.lines["cold"]`. I think it is neater in this case because then you can use loops to repeat the same actions over the two sets of data.